From 794f88478ca9a5829ee2ade630843ba82a8ddbf0 Mon Sep 17 00:00:00 2001
From: David Alvarez <alvdavi@amazon.com>
Date: Tue, 7 Jan 2020 23:31:35 +0000
Subject: [PATCH 1/4] 8227473: Improve gstreamer media support

---
 .../gst-plugins-bad/gst/aiff/aiffparse.c      | 24 +++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/modules/media/src/main/native/gstreamer/gstreamer-lite/gst-plugins-bad/gst/aiff/aiffparse.c b/modules/media/src/main/native/gstreamer/gstreamer-lite/gst-plugins-bad/gst/aiff/aiffparse.c
index d0da8a43..5e16fac6 100644
--- a/modules/media/src/main/native/gstreamer/gstreamer-lite/gst-plugins-bad/gst/aiff/aiffparse.c
+++ b/modules/media/src/main/native/gstreamer/gstreamer-lite/gst-plugins-bad/gst/aiff/aiffparse.c
@@ -643,7 +643,11 @@ gst_aiff_parse_calculate_duration (GstAiffParse * aiff)
   return FALSE;
 }
 
+#ifdef GSTREAMER_LITE
+static gboolean
+#else
 static void
+#endif
 gst_aiff_parse_ignore_chunk (GstAiffParse * aiff, guint32 tag, guint32 size)
 {
 #ifdef GSTREAMER_LITE
@@ -654,7 +658,11 @@ gst_aiff_parse_ignore_chunk (GstAiffParse * aiff, guint32 tag, guint32 size)
 
   if (aiff->streaming) {
     if (!gst_aiff_parse_peek_chunk (aiff, &tag, &size))
+#ifdef GSTREAMER_LITE
+      return FALSE;
+#else
       return;
+#endif
   }
   GST_WARNING_OBJECT (aiff, "Ignoring tag %" GST_FOURCC_FORMAT,
       GST_FOURCC_ARGS (tag));
@@ -667,6 +675,10 @@ gst_aiff_parse_ignore_chunk (GstAiffParse * aiff, guint32 tag, guint32 size)
   if (aiff->streaming) {
     gst_adapter_flush (aiff->adapter, flush);
   }
+
+#ifdef GSTREAMER_LITE
+  return TRUE;
+#endif
 }
 
 static double
@@ -1124,11 +1136,23 @@ gst_aiff_parse_stream_headers (GstAiffParse * aiff)
       }
       case GST_MAKE_FOURCC ('C', 'H', 'A', 'N'):{
         GST_FIXME_OBJECT (aiff, "Handle CHAN chunk with channel layouts");
+#ifdef GSTREAMER_LITE
+        if (!gst_aiff_parse_ignore_chunk (aiff, tag, size)) {
+          return GST_FLOW_OK;
+        }
+#else
         gst_aiff_parse_ignore_chunk (aiff, tag, size);
+#endif
         break;
       }
       default:
+#ifdef GSTREAMER_LITE
+        if (!gst_aiff_parse_ignore_chunk (aiff, tag, size)) {
+          return GST_FLOW_OK;
+        }
+#else
         gst_aiff_parse_ignore_chunk (aiff, tag, size);
+#endif
     }
 
     buf = NULL;
-- 
2.17.2

