diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/linux/config.h b/modules/web/src/main/native/Source/ThirdParty/libxml/linux/config.h
index 9a8a5775..2febd4f5 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/linux/config.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/linux/config.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /* config.h.  Generated from config.h.in by configure.  */
 /* config.h.in.  Generated from configure.ac by autoheader.  */

diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/linux/include/libxml/xmlversion.h b/modules/web/src/main/native/Source/ThirdParty/libxml/linux/include/libxml/xmlversion.h
index e3caf9eb..edc00d46 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/linux/include/libxml/xmlversion.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/linux/include/libxml/xmlversion.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: compile-time version informations
  * Description: compile-time version informations for the XML library
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/mac/config.h b/modules/web/src/main/native/Source/ThirdParty/libxml/mac/config.h
index ba5bc552..274ef4fb 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/mac/config.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/mac/config.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /* config.h.  Generated from config.h.in by configure.  */
 /* config.h.in.  Generated from configure.ac by autoheader.  */

diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/mac/include/libxml/xmlversion.h b/modules/web/src/main/native/Source/ThirdParty/libxml/mac/include/libxml/xmlversion.h
index e3caf9eb..edc00d46 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/mac/include/libxml/xmlversion.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/mac/include/libxml/xmlversion.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: compile-time version informations
  * Description: compile-time version informations for the XML library
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/HTMLparser.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/HTMLparser.c
index bf32db8f..c326228d 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/HTMLparser.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/HTMLparser.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * HTMLparser.c : an HTML 4.0 non-verifying parser
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/HTMLtree.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/HTMLtree.c
index 992bdf5b..5bc40b02 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/HTMLtree.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/HTMLtree.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * HTMLtree.c : implementation of access function for an HTML tree.
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/SAX2.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/SAX2.c
index d02fa194..3e426a16 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/SAX2.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/SAX2.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * SAX2.c : Default SAX2 handler to build a tree.
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/buf.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/buf.c
index 18c72965..dfa0d90d 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/buf.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/buf.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * buf.c: memory buffers for libxml2
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/chvalid.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/chvalid.c
index 326354bf..3857a5e9 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/chvalid.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/chvalid.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * chvalid.c:   this module implements the character range
  *      validation APIs
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/config.h.in b/modules/web/src/main/native/Source/ThirdParty/libxml/src/config.h.in
index 0b4aad67..df4c05bd 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/config.h.in
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/config.h.in
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /* config.h.in.  Generated from configure.ac by autoheader.  */

 /* Type cast for the gethostbyname() argument */
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/dict.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/dict.c
index 422a33d3..d3f3d6f3 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/dict.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/dict.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * dict.c: dictionary of reusable strings, just used to avoid allocation
  *         and freeing operations.
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/elfgcchack.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/elfgcchack.h
index 5ac93bfb..ebc6355c 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/elfgcchack.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/elfgcchack.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * elfgcchack.h: hack by Arjan van de Ven <arjanv@redhat.com> to speed
  *               up the code when using gcc for call within the library.
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/enc.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/enc.h
index 057d206d..20454b98 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/enc.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/enc.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Internal Interfaces for encoding in libxml2
  * Description: this module describes a few interfaces which were
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/encoding.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/encoding.c
index b014eb21..d5b73e50 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/encoding.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/encoding.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * encoding.c : implements the encoding conversion functions needed for XML
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/entities.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/entities.c
index 18ed507b..ff2e40ba 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/entities.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/entities.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * entities.c : implementation for the XML entities handling
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/error.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/error.c
index be255f10..87340f97 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/error.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/error.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * error.c: module displaying/handling XML parser errors
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/globals.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/globals.c
index d8ffd105..a5d2b63c 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/globals.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/globals.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * globals.c: definition and handling of the set of global variables
  *            of the library
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/hash.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/hash.c
index 5e369b95..5eaa79d3 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/hash.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/hash.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * hash.c: chained hash tables
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/DOCBparser.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/DOCBparser.h
index d7ed2bf7..29afaa51 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/DOCBparser.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/DOCBparser.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: old DocBook SGML parser
  * Description: interface for a DocBook SGML non-verifying parser
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/HTMLparser.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/HTMLparser.h
index b35c1021..ad110113 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/HTMLparser.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/HTMLparser.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for an HTML 4.0 non-verifying parser
  * Description: this module implements an HTML 4.0 non-verifying parser
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/HTMLtree.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/HTMLtree.h
index 6b87e83d..fe9ec6e5 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/HTMLtree.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/HTMLtree.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: specific APIs to process HTML tree, especially serialization
  * Description: this module implements a few function needed to process
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/SAX.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/SAX.h
index 3abc0b61..326615c5 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/SAX.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/SAX.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Old SAX version 1 handler, deprecated
  * Description: DEPRECATED set of SAX version 1 interfaces used to
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/SAX2.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/SAX2.h
index 6a30e421..4fda0450 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/SAX2.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/SAX2.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: SAX2 parser interface used to build the DOM tree
  * Description: those are the default SAX2 interfaces used by
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/c14n.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/c14n.h
index 5cdf6349..604d2b42 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/c14n.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/c14n.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Provide Canonical XML and Exclusive XML Canonicalization
  * Description: the c14n modules provides a
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/catalog.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/catalog.h
index c515c802..4d84ee0d 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/catalog.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/catalog.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /**
  * Summary: interfaces to the Catalog handling system
  * Description: the catalog module implements the support for
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/chvalid.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/chvalid.h
index 8d3df575..d1bb1681 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/chvalid.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/chvalid.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Unicode character range checking
  * Description: this module exports interfaces for the character
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/debugXML.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/debugXML.h
index 1750b645..ba67282d 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/debugXML.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/debugXML.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Tree debugging APIs
  * Description: Interfaces to a set of routines used for debugging the tree
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/dict.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/dict.h
index 60033641..02242dec 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/dict.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/dict.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: string dictionary
  * Description: dictionary of reusable strings, just used to avoid allocation
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/encoding.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/encoding.h
index 506bbb4d..a066de5d 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/encoding.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/encoding.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the encoding conversion functions
  * Description: interface for the encoding conversion functions needed for
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/entities.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/entities.h
index 4ba70391..d6322e76 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/entities.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/entities.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the XML entities handling
  * Description: this module provides some of the entity API needed
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/hash.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/hash.h
index 90cf1af5..730800a3 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/hash.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/hash.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Chained hash tables
  * Description: This module implements the hash table support used in
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/list.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/list.h
index 9cc6b0d6..363edfcd 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/list.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/list.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: lists interfaces
  * Description: this module implement the list support used in
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/nanoftp.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/nanoftp.h
index b7d80388..b13a7bb1 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/nanoftp.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/nanoftp.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: minimal FTP implementation
  * Description: minimal FTP implementation allowing to fetch resources
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/nanohttp.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/nanohttp.h
index 4f16851a..60d9680e 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/nanohttp.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/nanohttp.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: minimal HTTP implementation
  * Description: minimal HTTP implementation allowing to fetch resources
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/parser.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/parser.h
index 557dfc13..b6123571 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/parser.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/parser.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: the core parser module
  * Description: Interfaces, constants and types related to the XML parser
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/parserInternals.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/parserInternals.h
index 5e08bc48..9629e88a 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/parserInternals.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/parserInternals.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: internals routines and limits exported by the parser.
  * Description: this module exports a number of internal parsing routines
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/pattern.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/pattern.h
index d24b49af..2b73417f 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/pattern.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/pattern.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: pattern expression handling
  * Description: allows to compile and test pattern expressions for nodes
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/relaxng.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/relaxng.h
index 3d5bd80a..23fa1943 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/relaxng.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/relaxng.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: implementation of the Relax-NG validation
  * Description: implementation of the Relax-NG validation
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/schemasInternals.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/schemasInternals.h
index 7d59eb15..937a642c 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/schemasInternals.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/schemasInternals.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: internal interfaces for XML Schemas
  * Description: internal interfaces for the XML Schemas handling
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/schematron.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/schematron.h
index de76eb76..d3dcdb29 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/schematron.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/schematron.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: XML Schemastron implementation
  * Description: interface to the XML Schematron validity checking.
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/threads.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/threads.h
index 9cb9f8a5..5035e3e4 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/threads.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/threads.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /**
  * Summary: interfaces for thread handling
  * Description: set of generic threading related routines
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/tree.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/tree.h
index 7fe05067..df4160d7 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/tree.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/tree.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interfaces for tree manipulation
  * Description: this module describes the structures found in an tree resulting
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/uri.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/uri.h
index b96352a5..94fa176f 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/uri.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/uri.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /**
  * Summary: library of generic URI related routines
  * Description: library of generic URI related routines
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/valid.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/valid.h
index 4cb788a6..3ca79e59 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/valid.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/valid.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: The DTD validation
  * Description: API for the DTD handling and the validity checking
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xinclude.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xinclude.h
index e3e4ac91..1c758266 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xinclude.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xinclude.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: implementation of XInclude
  * Description: API to handle XInclude processing,
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xlink.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xlink.h
index 7f86f3d9..b1a735bc 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xlink.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xlink.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: unfinished XLink detection module
  * Description: unfinished XLink detection module
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlIO.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlIO.h
index 10c6ea26..46ae7813 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlIO.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlIO.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the I/O interfaces used by the parser
  * Description: interface for the I/O interfaces used by the parser
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlautomata.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlautomata.h
index 2925b767..70525f3e 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlautomata.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlautomata.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: API to build regexp automata
  * Description: the API to build regexp automata
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlerror.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlerror.h
index 04434a45..682efe4d 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlerror.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlerror.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: error handling
  * Description: the API used to report errors
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlexports.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlexports.h
index 2c79f814..b69d79d8 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlexports.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlexports.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: macros for marking symbols as exportable/importable.
  * Description: macros for marking symbols as exportable/importable.
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlmemory.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlmemory.h
index 7e073028..0c1a2654 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlmemory.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlmemory.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the memory allocator
  * Description: provides interfaces for the memory allocator,
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlmodule.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlmodule.h
index 1c72a7d2..c76393a4 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlmodule.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlmodule.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: dynamic module loading
  * Description: basic API for dynamic module loading, used by
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlreader.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlreader.h
index ae4c166d..6fb310e3 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlreader.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlreader.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: the XMLReader implementation
  * Description: API of the XML streaming API based on C# interfaces.
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlregexp.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlregexp.h
index 933c0cc1..b01ab2ef 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlregexp.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlregexp.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: regular expressions handling
  * Description: basic API for libxml regular expressions handling used
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlsave.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlsave.h
index e7ed8700..f0caef9b 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlsave.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlsave.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: the XML document serializer
  * Description: API to save document or subtree of document
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlschemas.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlschemas.h
index 4d9fc1bd..41aecf0a 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlschemas.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlschemas.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: incomplete XML Schemas structure implementation
  * Description: interface to the XML Schemas handling and schema validity
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlschemastypes.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlschemastypes.h
index 53e2377c..903fb351 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlschemastypes.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlschemastypes.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: implementation of XML Schema Datatypes
  * Description: module providing the XML Schema Datatypes implementation
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlunicode.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlunicode.h
index e77ea39a..5a877f44 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlunicode.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlunicode.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Unicode character APIs
  * Description: API for the Unicode character APIs
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlversion.h.in b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlversion.h.in
index b173be9e..0483cbc0 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlversion.h.in
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlversion.h.in
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: compile-time version informations
  * Description: compile-time version informations for the XML library
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlwriter.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlwriter.h
index a31cfac4..f1607092 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlwriter.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xmlwriter.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: text writing API for XML
  * Description: text writing API for XML
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpath.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpath.h
index a80a9f18..99ace34b 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpath.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpath.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: XML Path Language implementation
  * Description: API for the XML Path Language implementation
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpathInternals.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpathInternals.h
index 36cc18f6..e02d351f 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpathInternals.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpathInternals.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: internal interfaces for XML Path Language implementation
  * Description: internal interfaces for XML Path Language implementation
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpointer.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpointer.h
index 7cc45efa..ff984756 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpointer.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/libxml/xpointer.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: API to handle XML Pointers
  * Description: API to handle XML Pointers
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/private/regexp.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/private/regexp.h
new file mode 100644
index 00000000..b55c932b
--- /dev/null
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/include/private/regexp.h
@@ -0,0 +1,23 @@
+#ifndef XML_REGEXP_H_PRIVATE__
+#define XML_REGEXP_H_PRIVATE__
+
+#include <libxml/xmlautomata.h>
+
+#ifdef LIBXML_REGEXP_ENABLED
+
+/*
+ * -2 and -3 are used by xmlValidateElementType for other things.
+ */
+#define XML_REGEXP_OK               0
+#define XML_REGEXP_NOT_FOUND        (-1)
+#define XML_REGEXP_INTERNAL_ERROR   (-4)
+#define XML_REGEXP_OUT_OF_MEMORY    (-5)
+#define XML_REGEXP_INTERNAL_LIMIT   (-6)
+#define XML_REGEXP_INVALID_UTF8     (-7)
+
+XML_HIDDEN void
+xmlAutomataSetFlags(xmlAutomataPtr am, int flags);
+
+#endif /* LIBXML_REGEXP_ENABLED */
+
+#endif /* XML_REGEXP_H_PRIVATE__ */
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml-2.0-uninstalled.pc.in b/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml-2.0-uninstalled.pc.in
index 60b886b3..9ae39fbb 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml-2.0-uninstalled.pc.in
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml-2.0-uninstalled.pc.in
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 prefix=
 exec_prefix=
 libdir=${pcfiledir}
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml-2.0.pc.in b/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml-2.0.pc.in
index 2653a7c5..7649d820 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml-2.0.pc.in
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml-2.0.pc.in
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml.h
index 4fe56d21..976d9a4b 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * libxml.h: internal header only used during the compilation of libxml
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml.spec.in b/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml.spec.in
index 1996ccc4..d89ac6a1 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml.spec.in
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml.spec.in
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 %global with_python3 1

 Summary: Library providing XML and HTML support
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml2-config.cmake.in b/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml2-config.cmake.in
index 6b16fc2e..deb3bd00 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml2-config.cmake.in
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/libxml2-config.cmake.in
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 # libxml2-config.cmake
 # --------------------
 #
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/list.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/list.c
index 72a0e1c1..53b6516c 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/list.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/list.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * list.c: lists handling implementation
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/nanoftp.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/nanoftp.c
index cac81965..543ccea0 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/nanoftp.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/nanoftp.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * nanoftp.c: basic FTP client support
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/nanohttp.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/nanohttp.c
index ff64c461..e1a59bbc 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/nanohttp.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/nanohttp.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * nanohttp.c: minimalist HTTP GET implementation to fetch external subsets.
  *             focuses on size, streamability, reentrancy and portability
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/parser.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/parser.c
index c4e3ffff..6d0e2e29 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/parser.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/parser.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * parser.c : an XML 1.0 parser, namespaces and validity support are mostly
  *            implemented on top of the SAX interfaces
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/parserInternals.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/parserInternals.c
index 77787340..8b1d5aed 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/parserInternals.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/parserInternals.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * parserInternals.c : Internal routines (and obsolete ones) needed for the
  *                     XML and HTML parsers.
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/pattern.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/pattern.c
index 6ebd88ac..163da577 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/pattern.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/pattern.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * pattern.c: Implemetation of selectors for nodes
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/save.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/save.h
index 2c32a103..35855b35 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/save.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/save.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Internal Interfaces for saving in libxml2
  * Description: this module describes a few interfaces which were
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/threads.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/threads.c
index b443e1f3..4da1a9e8 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/threads.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/threads.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /**
  * threads.c: set of generic threading related routines
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/tree.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/tree.c
index 57d071d9..328ddbff 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/tree.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/tree.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * tree.c : implementation of access function for an XML tree.
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/triodef.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/triodef.h
index 46772ee2..e059947e 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/triodef.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/triodef.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*************************************************************************
  *
  * $Id$
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/trionan.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/trionan.c
index d5216aff..bde6ec59 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/trionan.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/trionan.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*************************************************************************
  *
  * $Id$
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/trionan.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/trionan.h
index 670075df..d1c7e6a2 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/trionan.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/trionan.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*************************************************************************
  *
  * $Id$
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/uri.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/uri.c
index 4cf4088c..18ac2d24 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/uri.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/uri.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /**
  * uri.c: set of generic URI related routines
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/valid.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/valid.c
index c9671f06..61aa277a 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/valid.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/valid.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * valid.c : part of the code use to do the DTD handling and the validity
  *           checking
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xml2Conf.sh.in b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xml2Conf.sh.in
index 08cb2334..f5e41230 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xml2Conf.sh.in
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xml2Conf.sh.in
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 #
 # Configuration file for using the XML library in GNOME applications
 #
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlIO.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlIO.c
index 4974d5fe..03c9717d 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlIO.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlIO.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * xmlIO.c : implementation of the I/O interfaces used by the parser
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlmemory.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlmemory.c
index e21cfd68..5e823f97 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlmemory.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlmemory.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * xmlmemory.c:  libxml memory allocator wrapper.
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlreader.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlreader.c
index f37d256e..92eb947f 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlreader.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlreader.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * xmlreader.c: implements the xmlTextReader streaming node API
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlsave.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlsave.c
index 3594e5d8..e0f24cfc 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlsave.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlsave.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * xmlsave.c: Implemetation of the document serializer
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlstring.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlstring.c
index 54e1ac41..e35f7620 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlstring.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlstring.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * string.c : an XML string utilities module
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlunicode.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlunicode.c
index 47553262..6e830087 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlunicode.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlunicode.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * xmlunicode.c: this module implements the Unicode character APIs
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlwriter.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlwriter.c
index db0de06d..193a1306 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlwriter.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xmlwriter.c
@@ -1,4 +1,8 @@

+#error "this file is unused"
+
+
+
 /*
  * xmlwriter.c: XML text writer implementation
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xpath.c b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xpath.c
index 2f8e4c1b..13ca9983 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xpath.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xpath.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * xpath.c: XML Path Language implementation
  *          XPath is a language for addressing parts of an XML document,
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xzlib.h b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xzlib.h
index 29ba55e7..885d430b 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxml/src/xzlib.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxml/src/xzlib.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /**
  * xzlib.h: header for the front end for the transparent suport of lzma
  *          compression at the I/O layer
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/linux/config.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/linux/config.h
index bd44c163..8ce59d6e 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/linux/config.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/linux/config.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /* config.h.  Generated from config.h.in by configure.  */
 /* config.h.in.  Generated from configure.ac by autoheader.  */

diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/mac/config.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/mac/config.h
index 65045345..2630e2dc 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/mac/config.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/mac/config.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /* config.h.  Generated from config.h.in by configure.  */
 /* config.h.in.  Generated from configure.ac by autoheader.  */

diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/config.h.in b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/config.h.in
index 085354e9..7547cae6 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/config.h.in
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/config.h.in
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /* config.h.in.  Generated from configure.ac by autoheader.  */

 /* Define to 1 if you have the <ansidecl.h> header file. */
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libexslt.pc.in b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libexslt.pc.in
index 16676ff6..556dc978 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libexslt.pc.in
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libexslt.pc.in
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt.pc.in b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt.pc.in
index 082d64cd..59a62f11 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt.pc.in
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt.pc.in
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attributes.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attributes.c
index fa5f3d9d..eb35c355 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attributes.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attributes.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * attributes.c: Implementation of the XSLT attributes handling
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attributes.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attributes.h
index 9b261ed0..1b608c70 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attributes.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attributes.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the XSLT attribute handling
  * Description: this module handles the specificities of attribute
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attrvt.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attrvt.c
index ca832e4b..2c1590c9 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attrvt.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attrvt.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * attrvt.c: Implementation of the XSL Transformation 1.0 engine
  *           attribute value template handling part.
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/documents.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/documents.c
index 6a7c954a..ca7730d8 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/documents.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/documents.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * documents.c: Implementation of the documents handling
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/documents.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/documents.h
index a8aa3a6f..bc0c8d97 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/documents.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/documents.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the document handling
  * Description: implements document loading and cache (multiple
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extensions.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extensions.c
index f8b01626..1d583d08 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extensions.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extensions.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * extensions.c: Implemetation of the extensions support
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extensions.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extensions.h
index d60145a5..0bf133bf 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extensions.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extensions.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the extension support
  * Description: This provide the API needed for simple and module
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extra.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extra.c
index 5c3c6ff9..a7e64bd0 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extra.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extra.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * extra.c: Implementation of non-standard features
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extra.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extra.h
index ce97d170..ac0ea9ca 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extra.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/extra.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the non-standard features
  * Description: implement some extension outside the XSLT namespace
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/functions.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/functions.c
index 3886684f..37d2c5d7 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/functions.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/functions.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * functions.c: Implementation of the XSLT extra functions
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/functions.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/functions.h
index 02938781..17e8174b 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/functions.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/functions.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the XSLT functions not from XPath
  * Description: a set of extra functions coming from XSLT but not in XPath
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/imports.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/imports.c
index 61d323b7..df4550af 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/imports.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/imports.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * imports.c: Implementation of the XSLT imports
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/imports.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/imports.h
index 89a1b940..e9a71c7d 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/imports.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/imports.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the XSLT import support
  * Description: macros and fuctions needed to implement and
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/keys.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/keys.c
index fe027b47..269fc5e9 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/keys.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/keys.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * keys.c: Implemetation of the keys support
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/keys.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/keys.h
index 4b2d7b69..eba2f24c 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/keys.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/keys.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary:  interface for the key matching used in key() and template matches.
  * Description: implementation of the key mechanims.
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/namespaces.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/namespaces.c
index b7fd56fd..631d82b5 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/namespaces.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/namespaces.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * namespaces.c: Implementation of the XSLT namespaces handling
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/namespaces.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/namespaces.h
index e2633c6c..15260a83 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/namespaces.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/namespaces.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the XSLT namespace handling
  * Description: set of function easing the processing and generation
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/numbers.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/numbers.c
index 5e15920b..a3b6dde1 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/numbers.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/numbers.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * numbers.c: Implementation of the XSLT number functions
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/numbersInternals.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/numbersInternals.h
index 613c1397..184e0de4 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/numbersInternals.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/numbersInternals.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Implementation of the XSLT number functions
  * Description: Implementation of the XSLT number functions
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/pattern.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/pattern.c
index 6ad651f9..0cc1f056 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/pattern.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/pattern.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * pattern.c: Implemetation of the template match compilation and lookup
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/pattern.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/pattern.h
index d8ddfa22..16b735a7 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/pattern.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/pattern.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the pattern matching used in template matches.
  * Description: the implementation of the lookup of the right template
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/preproc.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/preproc.c
index e9388de1..58e9e20c 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/preproc.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/preproc.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * preproc.c: Preprocessing of style operations
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/preproc.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/preproc.h
index 919a094b..58af6295 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/preproc.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/preproc.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: precomputing stylesheets
  * Description: this is the compilation phase, where most of the
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/security.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/security.c
index 13d477e2..958dcf12 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/security.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/security.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * security.c: Implementation of the XSLT security framework
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/security.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/security.h
index 724bb19b..bbc049b5 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/security.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/security.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the libxslt security framework
  * Description: the libxslt security framework allow to restrict
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/templates.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/templates.c
index 882d757e..e00c6b45 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/templates.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/templates.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * templates.c: Implementation of the template processing
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/templates.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/templates.h
index 04e72702..c18dfeeb 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/templates.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/templates.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the template processing
  * Description: This set of routine encapsulates XPath calls
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/transform.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/transform.c
index eb051c5a..b60c3bd6 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/transform.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/transform.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * transform.c: Implementation of the XSL Transformation 1.0 engine
  *              transform part, i.e. applying a Stylesheet to a document
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/transform.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/transform.h
index 2ea1d7d3..aeb11b2c 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/transform.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/transform.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: the XSLT engine transformation part.
  * Description: This module implements the bulk of the actual
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/trio.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/trio.h
index 8b5ebd4d..9ff215c5 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/trio.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/trio.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*************************************************************************
  *
  * $Id$
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/variables.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/variables.c
index 4c89d127..aacf10b8 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/variables.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/variables.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * variables.c: Implementation of the variable storage and lookup
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/variables.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/variables.h
index 8f1837fe..a7223f80 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/variables.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/variables.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: interface for the variable matching and lookup.
  * Description: interface for the variable matching and lookup.
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/win32config.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/win32config.h
index 487962a6..c25ba111 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/win32config.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/win32config.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Windows configuration header
  * Description: Windows configuration header
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xslt.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xslt.c
index 8f5addfd..8b77e568 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xslt.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xslt.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * xslt.c: Implemetation of an XSL Transformation 1.0 engine
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xslt.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xslt.h
index e8191c9b..855a3242 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xslt.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xslt.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Interfaces, constants and types related to the XSLT engine
  * Description: Interfaces, constants and types related to the XSLT engine
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltInternals.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltInternals.h
index 20f34fc6..146f3510 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltInternals.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltInternals.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: internal data structures, constants and functions
  * Description: Internal data structures, constants and functions used
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltconfig.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltconfig.h
index 3980eb13..1246425d 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltconfig.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltconfig.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: compile-time version informations for the XSLT engine
  * Description: compile-time version informations for the XSLT engine
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.c
index 63fbaa66..e1ef2e91 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * xsltlocale.c: locale handling
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h
index 5253de72..d2ad564b 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: Locale handling
  * Description: Interfaces for locale handling. Needed for language dependent
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltutils.c b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltutils.c
index 840f2959..772d00b4 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltutils.c
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltutils.c
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * xsltutils.c: Utilities for the XSL Transformation 1.0 engine
  *
diff --git a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltutils.h b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltutils.h
index c94e51b8..ddec1179 100644
--- a/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltutils.h
+++ b/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltutils.h
@@ -1,3 +1,7 @@
+
+#error "this file is unused"
+
+
 /*
  * Summary: set of utilities for the XSLT engine
  * Description: interfaces for the utilities module of the XSLT engine.
