diff --git a/bin/rt-8u202-ga/modules/web/src/main/native/Source/ThirdParty/sqlite/sqlite3.c b/bin/rt-8u202-ga/modules/web/src/main/native/Source/ThirdParty/sqlite/sqlite3.c
index b42077bb..e6b1974f 100644
--- a/bin/rt-8u202-ga/modules/web/src/main/native/Source/ThirdParty/sqlite/sqlite3.c
+++ b/bin/rt-8u202-ga/modules/web/src/main/native/Source/ThirdParty/sqlite/sqlite3.c
@@ -1,3 +1,6 @@
+#error "this file is unused"
+
+
 /******************************************************************************
 ** This file is an amalgamation of many separate C source files from SQLite
 ** version 3.23.0.  By combining all the individual C code files into this
diff --git a/bin/rt-8u202-ga/modules/web/src/main/native/Source/ThirdParty/sqlite/sqlite3.h b/bin/rt-8u202-ga/modules/web/src/main/native/Source/ThirdParty/sqlite/sqlite3.h
index 506f2a8c..68d7a65f 100644
--- a/bin/rt-8u202-ga/modules/web/src/main/native/Source/ThirdParty/sqlite/sqlite3.h
+++ b/bin/rt-8u202-ga/modules/web/src/main/native/Source/ThirdParty/sqlite/sqlite3.h
@@ -1,3 +1,6 @@
+#error "this file is unused"
+
+
 /*
 ** 2001-09-15
 **
