--- a/buildSrc/mac.gradle	2018-10-26 13:52:53.000000000 -0700
+++ b/buildSrc/mac.gradle	2018-10-25 13:58:57.000000000 -0700
@@ -51,7 +51,7 @@

 // Set the minimum API version that we require (developers do not need to override this)
 // Note that this is not necessarily the same as the preferred SDK version
-defineProperty("MACOSX_MIN_VERSION", "10.10");
+defineProperty("MACOSX_MIN_VERSION", "10.8");

 // Create $buildDir/mac_tools.properties file and load props from it
 setupTools("mac_tools",
