From e987418ad924166baff4a569b8f948f3dcffc739 Mon Sep 17 00:00:00 2001
From: Philipp Trulson <philipp@trulson.de>
Date: Fri, 1 Feb 2019 15:54:08 +0100
Subject: [PATCH 09/12] Disable compiling errors

---
 buildSrc/linux.gradle | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/buildSrc/linux.gradle b/buildSrc/linux.gradle
index 184fa55..e0c14bb 100644
--- a/buildSrc/linux.gradle
+++ b/buildSrc/linux.gradle
@@ -206,7 +206,7 @@ LINUX.glass.javahInclude = [
     "com/sun/glass/ui/gtk/*"]
 LINUX.glass.glass.nativeSource = ft_gtk_launcher.getFiles()
 LINUX.glass.glass.compiler = compiler
-LINUX.glass.glass.ccFlags = [ccFlags, gtk2CCFlags,  "-Werror"].flatten()
+LINUX.glass.glass.ccFlags = [ccFlags, gtk2CCFlags].flatten()
 LINUX.glass.glass.linker = linker
 LINUX.glass.glass.linkFlags = [linkFlags, "-lX11", "-ldl" ].flatten()
 LINUX.glass.glass.lib = "glass"
@@ -214,7 +214,7 @@ LINUX.glass.glass.lib = "glass"
 LINUX.glass.glassgtk2 = [:]
 LINUX.glass.glassgtk2.nativeSource =  ft_gtk.getFiles()
 LINUX.glass.glassgtk2.compiler = compiler
-LINUX.glass.glassgtk2.ccFlags = [ccFlags, gtk2CCFlags, "-Werror"].flatten()
+LINUX.glass.glassgtk2.ccFlags = [ccFlags, gtk2CCFlags].flatten()
 LINUX.glass.glassgtk2.linker = linker
 LINUX.glass.glassgtk2.linkFlags = [linkFlags, gtk2LinkFlags ].flatten()
 LINUX.glass.glassgtk2.lib = "glassgtk2"
@@ -222,7 +222,7 @@ LINUX.glass.glassgtk2.lib = "glassgtk2"
 LINUX.glass.glassgtk3 = [:]
 LINUX.glass.glassgtk3.nativeSource =  ft_gtk.getFiles()
 LINUX.glass.glassgtk3.compiler = compiler
-LINUX.glass.glassgtk3.ccFlags = [ccFlags, gtk3CCFlags, "-Werror"].flatten()
+LINUX.glass.glassgtk3.ccFlags = [ccFlags, gtk3CCFlags].flatten()
 LINUX.glass.glassgtk3.linker = linker
 LINUX.glass.glassgtk3.linkFlags = [linkFlags, gtk3LinkFlags ].flatten()
 LINUX.glass.glassgtk3.lib = "glassgtk3"
-- 
1.8.3.1

